### ISSUE TITLE
SIGSEGV in go vet

### ISSUE THREAD

**Issue by kolyshkin** [golang/go#21794]
I am seeing the following while running `go vet` (via `gometalinter`) on Moby sources.

### What version of Go are you using (`go version`)?
go version go1.8.3 linux/arm

### Does this issue reproduce with the latest release?
Haven't tried yet

### What operating system and processor architecture are you using (`go env`)?
```
GOARCH="arm"
GOBIN=""
GOEXE=""
GOHOSTARCH="arm"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/go"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/linux_arm"
GCCGO="gccgo"
GOARM="7"
CC="gcc"
GOGCCFLAGS="-fPIC -marm -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build308764312=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
PKG_CONFIG="pkg-config"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
```
(a Scaleway C1 machine running official Docker image)

### What did you do?
```
git clone https://github.com/moby/moby
cd moby
make all
```

### What did you expect to see?
No SIGSEGV from go vet

### What did you see instead?

```
DEBUG: warning: /usr/local/go/bin/go returned exit status 1: panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1f593c]

goroutine 1 [running]:
main.checkLostCancel(0x108b8180, 0x371228, 0x108b3e40)
	/usr/local/go/src/cmd/vet/lostcancel.go:107 +0x58c
main.(*File).Visit(0x108b8180, 0x371228, 0x108b3e40, 0x3708c0, 0x108b8180)
	/usr/local/go/src/cmd/vet/main.go:500 +0xdc
go/ast.Walk(0x3708c0, 0x108b8180, 0x371228, 0x108b3e40)
	/usr/local/go/src/go/ast/walk.go:52 +0x38
go/ast.walkDeclList(0x3708c0, 0x108b8180, 0x108b8100, 0x10, 0x10)
	/usr/local/go/src/go/ast/walk.go:38 +0x6c
go/ast.Walk(0x3708c0, 0x108b8180, 0x371208, 0x10863d00)
	/usr/local/go/src/go/ast/walk.go:353 +0x24a8
main.(*File).walkFile(0x108b8180, 0xbea421c6, 0x23, 0x10863d00)
	/usr/local/go/src/cmd/vet/main.go:467 +0xc0
main.doPackage(0x26ab10, 0x1, 0x1060e488, 0x8f, 0x8f, 0x0, 0xffffffff)
	/usr/local/go/src/cmd/vet/main.go:358 +0x8e4
main.main()
	/usr/local/go/src/cmd/vet/main.go:247 +0x3c0
```

**Comment by mvdan** [comment]
Have you tried Go 1.9?

**Comment by mvdan** [comment]
Yup, a duplicate of an issue already fixed in 1.9: https://github.com/golang/go/issues/19656.

Please update.
